PyInstaller/hooks/rthooks/pyi_rth_pkgres.py

```diff
@@ -9,14 +9,179 @@
+        # correct prefix in case of a module)
+        pkg_name = '.'.join(rel_pkg_path.parts)
+
+        # Collect relevant entries from TOC. We are interested in either
+        # files that are located in the package/module's directory (data
+        # files) or in packages that are prefixed with package/module's
+        # name (to reconstruct subpackage directories)
+        data_files = []
+        package_dirs = []
+
+        for entry in self.loader.toc:
+            entry_path = pathlib.PurePath(entry)
+
+            if rel_pkg_path in entry_path.parents:
+                # Data file path
+                data_files.append(entry_path)
+            elif entry.startswith(pkg_name) and self.loader.is_package(entry):
+                # Package or subpackage; convert the name to directory path
+                package_dir = pathlib.PurePath(*entry.split('.'))
+                package_dirs.append(package_dir)
+
+        # Reconstruct the filesystem
+        self.embedded_tree = _TocFilesystem(data_files, package_dirs)
+
+    def _normalize_path(self, path):
+        # Avoid using Path.resolve(), because it resolves symlinks. This
+        # is undesirable, because the pure path in self._pkg_path does
+        # not have symlinks resolved, so comparison between the two
+        # would be faulty. So use os.path.abspath() instead to normalize
+        # the path
+        return pathlib.Path(os.path.abspath(path))
+
+    def _is_relative_to_package(self, path):
+        return path == self._pkg_path or self._pkg_path in path.parents
+
+    def _has(self, path):
+        # Prevent access outside the package
+        path = self._normalize_path(path)
+        if not self._is_relative_to_package(path):
+            return False
+
+        # Relative path for searching embedded resources
+        rel_path = path.relative_to(SYS_PREFIX)
+
+        return self.embedded_tree.path_exists(rel_path) or path.exists()
+
+    def _isdir(self, path):
+        # Prevent access outside the package
+        path = self._normalize_path(path)
+        if not self._is_relative_to_package(path):
+            return False
+
+        # Relative path for searching embedded resources
+        rel_path = path.relative_to(SYS_PREFIX)
+
+        return self.embedded_tree.path_isdir(rel_path) or path.is_dir()
+
+    def _listdir(self, path):
+        # Prevent access outside the package
+        path = self._normalize_path(path)
+        if not self._is_relative_to_package(path):
+            return []
+
+        # Relative path for searching embedded resources
+        rel_path = path.relative_to(SYS_PREFIX)
+
+        # List content from embedded filesystem...
+        content = self.embedded_tree.path_listdir(rel_path)
+
+        # ... as well as the actual one
+        if path.is_dir():
+            # Use os.listdir() to avoid having to convert Path objects
+            # to strings...
+            path = str(path)  # python 3.5
+            content += os.listdir(path)
```

Comment:
Hmm, indeed. And I suspect this may already be happening for some of the directories, but we're missing that in the test because `set()` is used to compare the results. Will fix both...